models/User.go

```diff
@@ -2357,6 +2357,64 @@ func GetReservedStocksOwned(userId uint32) (map[uint32]int64, error) {
 	return reservedStocksOwned, nil
 }
 
+func GetCashSpent(userId uint32) (map[uint32]int64, error) {
+	var l = logger.WithFields(logrus.Fields{
+		"method": "GetCashSpent",
+		"userId": userId,
+	})
+
+	l.Info("GetCashSpent requested")
+
+	db := getDB()
+
+	sql := "Select stockId, sum(reservedStockQuantity*-1) as soldstockstotal from Transactions where userId=? and stockQuantity=0 and type='OrderFillTransaction' group by stockId"
+	rows, err := db.Raw(sql, userId).Rows()
+	if err != nil {
+		l.Error(err)
+		return nil, err
+	}
+	defer rows.Close()
+
+	soldStocksTotal := make(map[uint32]int64)
+	for rows.Next() {
+		var stockId uint32
+		var soldstocks int64
+		rows.Scan(&stockId, &soldstocks)
+
+		soldStocksTotal[stockId] = soldstocks
+	}
+
+	sql1 := "Select stockId, stockQuantity, price from Transactions where userId=? and stockQuantity>0 and (type='OrderFillTransaction' or type='FromExchangeTransaction') order by stockId;"
```

Comment:
order by `stockId` is unnecessary, better to order by `createdAt` as we're using `FIFO` algorithm